galleries/examples/lines_bars_and_markers/color_line_by_value.py

```diff
@@ -0,0 +1,95 @@
+"""
+===================
+Color line by value
+===================
+
+Create a line where the color at each pointed is determined
+by a third value (in addition to x and y).
+"""
+
+import warnings
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from matplotlib.collections import LineCollection
+
+
+def colored_line(x, y, c, ax, **lc_kwargs):
+    """
+    Plot a line with a color specified along the line by a third value. It does this
+    by creating a collection of line segments. Each line segment is made up of two
+    straight lines each connecting the current (x, y) point to the midpoints of the
+    lines connecting the current point with its two neighbors. This creates a smooth
+    line with no gaps between the line segments.
```

Comment:
This is very minor, but I suggest that you add a new line to reflect standard docstring conventions.

```suggestion
def colored_line(x, y, c, ax, **lc_kwargs):
    """
    Plot a line with a color specified along the line by a third value.
    
    It does this by creating a collection of line segments. Each line segment
    is made up of two straight lines each connecting the current (x, y) point
    to the midpoints of the lines connecting the current point with its two
    neighbors. This creates a smooth line with no gaps between the line
    segments.
```